src/ScalarMessaging.js

```diff
@@ -0,0 +1,171 @@
+        _error: <Original Error Object>
+    }
+}
+The "message" key should be a human-friendly string.
+
+The response object for "membership_state" looks like:
+{
+    membership_state: "join" | "leave" | "invite" | "ban"
+}
+
+The response object for "invite" looks like:
+{
+    invite: true
+}
+
+*/
+
+const SdkConfig = require('./SdkConfig');
+const MatrixClientPeg = require("./MatrixClientPeg");
+
+const onMessage = function(event) {
+    if (!event.origin) { // stupid chrome
+        event.origin = event.originalEvent.origin;
+    }
+
+    // check it is from the integrations UI URL (remove trailing spaces)
+    let url = SdkConfig.get().integrations_ui_url;
+    if (url.endsWith("/")) {
+        url = url.substr(0, url.length - 1);
+    }
+    if (url !== event.origin) {
+        console.warn("Unauthorised postMessage received. Source URL: " + event.origin);
+        return;
+    }
+
+    switch (event.data.action) {
+        case "membership_state": {
+            const roomId = event.data.room_id;
+            const userId = event.data.user_id;
+            if (!userId) {
+                return sendError(event, "Missing user_id in request");
+            }
+            if (!roomId) {
+                return sendError(event, "Missing room_id in request");
+            }
+            console.log(`membership_state of ${userId} in room ${roomId} requested.`);
+            const client = MatrixClientPeg.get();
+            if (!client) {
+                return sendError(event, "You need to be logged in.");
+            }
+            const room = client.getRoom(roomId);
+            if (!room) {
+                return sendError(event, "This room is not recognised.");
+            }
+            let membershipState = "leave";
+            const member = room.getMember(userId);
+            if (member) {
+                membershipState = member.membership;
+            }
+            sendResponse(event, {
+                membership_state: membershipState
+            });
+        }
+            break;
+        case "invite": {
+            const roomId = event.data.room_id;
+            const userId = event.data.user_id;
+            if (!userId) {
+                return sendError(event, "Missing user_id in request");
+            }
+            if (!roomId) {
+                return sendError(event, "Missing room_id in request");
+            }
+            console.log(`Received request to invite ${userId} into room ${roomId}`);
+            const client = MatrixClientPeg.get();
+            if (!client) {
+                return sendError(event, "You need to be logged in.");
+            }
+            const room = client.getRoom(roomId);
+            if (room) {
+                // if they are already invited we can resolve immediately.
+                const member = room.getMember(userId);
+                if (member && member.membership === "invite") {
+                    sendResponse(event, {
```

Comment:
`sendError` doesn't actually return anything (and the function itself doesn't expect a return value. The reason why I in-lined it was because I was constantly forgetting to put `return;` on a new line after the `sendError`. Will make it multi-line again since that is more semantically correct.